vimeo/stag-java
# [BUG] Stag.Factory does not support reuse and should tell the consumer so

#### Ticket
https://github.com/vimeo/stag-java/issues/112

#### Ticket Summary
JSON was not properly parsed when trying to reuse a stag.factory instance between gson instances.

#### Implementation Summary
The chosen solution was to throw an `UnsupportedOperationException` when the consumer tries to reuse a `Stag.Factory` between `Gson` instances. See the linked ticket for more discussion. Due to the nature of this change, I am proposing that the next version of stag be `2.3.0` rather than `2.2.1`. The reasoning is that this could cause crashes for consumers that are reusing the factory between instances. In the worst case, consumers unaffected by the above issue (due to never registering different factories) will have to change their code to use a new instance instead of reusing it. I believe this is acceptable enough to do, let me know if you disagree.

#### How to Test
Run the added unit test to ensure that an exception is thrown. Note that the unit test will be able to test equality in a future where factory instances can be reused.
